lib/applications/send.js

```diff
@@ -0,0 +1,189 @@
+'use strict';
+
+const Promise = require('bluebird');
+const moment = require('moment');
+const request = require('request');
+
+const database = require('../database');
+
+/*
+ * @typedef requestEndpoint
+ * @type object
+ *
+ * @property {array} options - options for the Request module
+ * @property {string} options.url - application endpoint url
+ * @property {string} options.method - type of request
+ * @property {object} options.json - request body
+ * @property {string} options.json.id - client id
+ * @property {string} options.json.secret - client secret
+ */
+
+/*
+ * @typedef endpointResponse
+ * @type object
+ * Contains the response and timestamp for an endpoint request
+ *
+ * @property {date} timestamp - timestamp of request
+ * @property {number} response - statusCode response from request
+ */
+
+/*
+ * @typedef options
+ * @type object
+ * Application options is the object passed between each of these functions
+ *
+ * @property {string} trigger - one of live/sunset/updated
+ * @property {array} apps - applications pulled from database
+ * @property {array} endpoints - endpoints derived from apps (first added by `endpoints` function)
+ * @property {string} endpoints.id - id of a given app (added by `endpoints` function)
+ * @property {requestEndpoint} endpoints.options - see requestEndpoint typedef (added by `endpoints` function)
+ * @property {endpointResponse} endpoints.response - see endpointResponse typedef (added during `send` Promise.resolve)
+ */
+
+/**
+ * Creates an array from all application endpoints that match the trigger
+ *
+ * @param  {options} options - see options typedef
+ *
+ * @returns {options}  - see options typedef
+ */
+const endpoints = options => {
+  const opts = options;
+
+  return new Promise((resolve, reject) => {
+    if (!Array.isArray(options.apps)) {
+      reject('Apps must be an array');
+    }
+
+    opts.endpoints = options.apps.map(app => {
+      if (app[`${options.trigger}-endpoint`] && app[`${options.trigger}-endpoint`] !== '') {
+        return {
+          id: app.id,
+          options: {
+            url: app[`${options.trigger}-endpoint`],
+            method: 'POST',
+            json: {
+              id: app['client-id'],
+              secret: app['client-secret'],
+            },
+          },
+        };
+      }
+
+      return null;
+    }).filter(Boolean);
```

Comment:
This is now the second way I've seen you filter out things that are `null` in this PR. Can we stick with a single method?